File: chor-extraction/src/main/java/ast/cc/CCVisitor.java
package ast.cc;

import ast.cc.nodes.*;
import epp.MergingException;

/**
 * Created by fmontesi on 03/04/17.
 */
public interface CCVisitor<T> {
    public T visit(Selection n) throws MergingException;

    public T visit(Communication n) throws MergingException;

    public T visit(Condition n) throws MergingException;

    public T visit(Termination n) throws MergingException;

    public T visit(ProcedureDefinition n) throws MergingException;

    public T visit(ProcedureInvocation n) throws MergingException;

    public T visit(Program n) throws MergingException;

    public T visit(BadTermination n);
}

File: pe/src/main/kotlin/ast/cc/CCVisitor.kt
package ast.cc

import ast.cc.nodes.*

/**
 * Created by fmontesi on 03/04/17.
 */
interface CCVisitor<T> {
    fun visit(n: Condition): T

    fun visit(n: Termination): T

    fun visit(n: ProcedureDefinition): T

    fun visit(n: ProcedureInvocation): T

    fun visit(n: Choreography): T

    fun visit(n: Program): T

    fun visit(n: Multicom): T

    fun visit(n: CommunicationSelection): T
}
